membrane_curvature/tests/test_membrane_curvature.py

```diff
@@ -129,6 +129,104 @@ def small_grofile():
     return sel
 
 
+def parametric_hemisphere(
+    radius: float, nx: int, ny: int, scale: float = 5
+) -> tuple[np.ndarray, float, float]:
+    """Generate the height field and analytical mean/gaussian curvature of a 
+    hemisphere given a sampling range. The range of the patch is defined by 
+    the hemisphere radius and a scale.
+
+    Parameters
+    ----------
+        radius: float
+            Radius of hemisphere
+        nx: int
+            Number of x gridpoints
+        ny: int 
+            Number of y grid points
+        patch_scale: float, optional
+            Scale factor for range of grid. Defaults to 5.
+
+    Returns
+    -------
+        Tuple[np.ndarray, float, float, float, float]
+            height field, mean curvature, gaussian curvature, dx, dy
+    """
+    x, dx = np.linspace(-radius / scale, radius / scale, nx, retstep=True)
+    y, dy = np.linspace(-radius / scale, radius / scale, ny, retstep=True)
+    xx, yy = np.meshgrid(x, y, indexing="ij")
+    zz = np.sqrt(radius**2 - xx**2 - yy**2)
+    H = -1 / radius
+    K = 1 / radius**2
+    return zz, H, K, dx, dy
+
+
+def test_parametric_hemisphere_curvature():
+    for radius in np.arange(5, 100, 5):
+        for nx in [7, 23, 47, 67]:
+            for ny in [7, 23, 47, 67]:
+                # Ensure that patch_scale is sufficiently large such that points are closely spaced
+                height_field, H, K, dx, dy = parametric_hemisphere(
+                    radius=radius, nx=nx, ny=ny, scale=200
+                )
+                mean = mean_curvature(height_field, dx, dy)
```

Comment:
Here `dx`  and `dy` are provided in a different order than what is defined in   [membrane_curvature/base.py](https://github.com/MDAnalysis/membrane-curvature/pull/116/files#diff-f8ce9f8ff47f05c29737ef8761516649aae370179a681477e6c37421e569a27b) . I'm failing to see why the order is different. Could you please clarify or document why is this the case?

Reply:
I got confused by the ordering of the output of np.gradient calls in curvature.py. 

```python
dy, dx = np.gradient(Z)
```

suggesting that the y-axis is the first axis etc... The first axis is I think the x-axis. It's a bit weird but it seems due to row-major ordering numpy arrays are contiguous in y but not x. I've gone ahead and swapped the order of outputs of np.gradient in a new commit.